src/ostree/ot-builtin-commit.c

```diff
@@ -147,13 +149,68 @@ parse_statoverride_file (GHashTable   **out_mode_add,
   return ret;
 }
 
+/* XXX: share more code with parse_statoverride_file() */
+static gboolean
+parse_skiplist_file (GHashTable   **out_skiplist,
+                     GCancellable  *cancellable,
+                     GError        **error)
+{
+  gboolean ret = FALSE;
+  gsize len;
+  char **iter = NULL; /* nofree */
+  g_autoptr(GHashTable) ret_hash = NULL;
+  g_autoptr(GFile) path = NULL;
+  g_autofree char *contents = NULL;
+  char **lines = NULL;
+
+  path = g_file_new_for_path (opt_skiplist_file);
+
+  if (!g_file_load_contents (path, cancellable, &contents, &len, NULL, error))
+    goto out;
+
+  ret_hash = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
+  lines = g_strsplit (contents, "\n", -1);
```

Comment:
I started working on doing this but felt silly doing it, so I went back to using `g_strsplit()`. It just felt like the trade-off in legibility was not worth it for something that people should probably use the API for if it somehow becomes a bottleneck.